packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -11,41 +16,155 @@ import icons from './icon-assets/icons.json';
 })
 export class Icon {
   /**
-   * Identifier of the icon. The attribute’s value must be a valid svg file in the repo of icons (assets/icons).
+   * Identifier of the icon. Type values 'registry' or 'icon'.
+   */
+  @Prop() type = 'icon';
+  /**
+   * Identifier of the icon. The attribute’s value must be a valid svg Name in the Crayons-Icon set.
    */
   @Prop() name: string;
-
   /**
-   * Size of the icon, specified in number of  pixels.
-   * Default value is 12px defined using the --icon-size css variable.
+   * Identifier of the icon. The attribute’s value must be a valid JS Import Name of the svg in the named export from @freshworks/crayons-icon.
+   */
+  @Prop() dataSvg = '';
+  /**
+   * Valid for fw-icon-type - registry only. This sets a context setter object which can then be used to render related icon from libraries.
+   */
+  @Prop() dataRegisterLibs: any;
```

Comment:
Any reason why this is a public prop?
Do you expect developers to set anything in this prop?